libsolidity/ExpressionCompiler.cpp

```diff
@@ -855,8 +857,21 @@ bool ExpressionCompiler::visit(IndexAccess const& _indexAccess)
 		solAssert(_indexAccess.getIndexExpression(), "Index expression expected.");
 
 		_indexAccess.getIndexExpression()->accept(*this);
+
```

Comment:
Please do this check as part of the type-checking phase, no the code generation phase. I.e. it probably has to go AST.cpp.